sys/linux/rdma.txt

```diff
@@ -0,0 +1,1464 @@
+# =======
+
+# context
+# ===========
+
+mlx5_get_context_cmd_resp {
+	async_fd			int32
+	num_comp_vectors		int32
+	qp_tab_size			int32
+	bf_reg_size			int32
+	tot_uuars			int32
+	cache_line_size			int32
+	max_sq_desc_sz			int16
+	max_rq_desc_sz			int16
+	max_send_wqebb			int32
+	max_recv_wr			int32
+	max_srq_recv_wr			int32
+	num_ports			int16
+	reserved1			const[0x0, int16]
+	comp_mask			int32[0:1]
+	response_length			int32
+	cqe_version_cmds_supp_uhw	int16
+	reserved2			const[0x0, int16]
+	hca_core_clock_offset		int64
+	log_uar_size			int32
+	num_uars_per_page		int32
+	num_dyn_bfregs			int32
+	reserved3			const[0x0, int32]
+}
+
+mlx5_get_context_cmd {
+	command			const[0x0, int32]
+	in_words		const[0xc, int16]
+	out_words		const[0x12, int16]
+	response		ptr[out, mlx5_get_context_cmd_resp]
+	total_num_uuars		int32
+	num_low_latency_uuars	int32
+	flags			int32
+	comp_mask		int32
+	cqe_version		int16
+	reserved1		const[0x0, int16]
+	reserved2		const[0x0, int32]
+	lib_caps		int64
+}
+
+# query_device
+# ================
+
+query_device_resp_ex {
+	fw_ver				int64
+	node_guid			int64
+	sys_image_guid			int64
+	max_mr_size			int64
+	page_size_cap			int64
+	vendor_id			int32
+	vendor_part_id			int32
+	hw_ver				int32
+	max_qp				int32
+	max_qp_wr			int32
+	device_cap_flags		int32
+	max_sge				int32
+	max_sge_rd			int32
+	max_cq				int32
+	max_cqe				int32
+	max_mr				int32
+	max_qp_rd_atom			int32
+	max_ee_rd_atom			int32
+	max_res_rd_atom			int32
+	max_qp_init_rd_atom		int32
+	max_ee_init_rd_atom		int32
+	atomic_cap			int32
+	max_ee				int32
+	max_rdd				int32
+	max_mw				int32
+	max_raw_ipv6_qp			int32
+	max_raw_ethy_qp			int32
+	max_mcast_grp			int32
+	max_mcast_qp_attach		int32
+	max_total_mcast_qp_attach	int32
+	max_ah				int32
+	max_fmr				int32
+	max_map_per_fmr			int32
+	max_srq				int32
+	max_srq_wr			int32
+	max_srq_sge			int32
+	max_pkeys			int16
+	local_ca_ack_delay		int8
+	phys_port_cnt			int8
+	reserved			const[0x0, int32]
+	comp_mask			const[0x0, int32]
+	response_length			int32
+	general_caps			int64
+	rc_odp_caps			int32
+	uc_odp_caps			int32
+	ud_odp_caps			int32
+	reserved_odp			const[0x0, int32]
+	timestamp_mask			int64
+	hca_core_clock			int64
+	device_cap_flags_ex		int64
+	supported_qpts			int32
+	max_rwq_indirection_tables	int32
+	max_rwq_indirection_table_size	int32
+	reserved_rss			const[0x0, int32]
+	max_wq_type_rq			int32
+	raw_packet_caps			int32
+	max_rndv_hdr_size		int32
+	max_num_tags			int32
+	flags				int32
+	max_ops				int32
+	max_sge_tm			int32
+	reserved_tm			int32
+	max_cq_moderation_count		int16
+	max_cq_moderation_period	int16
+	reserved_cq_mod			int32
+	max_dm_size			int64
+}
+
+query_device_cmd_ex {
+	command			const[IB_USER_VERBS_EX_CMD_QUERY_DEVICE, int32]
+	in_words		const[0x1, int16]
+	out_words		bytesize4[response, int16]
+	response		ptr[out, query_device_resp_ex]
+	provider_in_words	const[0x0, int16]
+	provider_out_words	const[0x8, int16]
+	reserved		const[0x0, int32]
+	comp_mask		const[0x0, int32]
```

Comment:
It seems these reserved and comp_mask fields correspond to ib_uverbs_ex_query_device struct:
https://elixir.bootlin.com/linux/v5.1-rc1/source/include/uapi/rdma/ib_user_verbs.h#L219
Please declare this struct explicitly with the name matching kernel definition.
It makes working with these description way easier. And in future we will have static checking for typos and bugs in descriptions:
https://github.com/google/syzkaller/issues/590